src/backend/sev/snp/firmware/mod.rs

```diff
@@ -0,0 +1,395 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Operations for managing the SEV platform.
+
+#[cfg(target_os = "linux")]
+mod linux;
+
+#[cfg(target_os = "linux")]
+pub use linux::Firmware;
+
+use super::Version;
+
+use bitflags::bitflags;
+
+use std::fmt::{Debug, Display, Formatter};
+use std::{error, io};
+
+/// There are a number of error conditions that can occur between this
+/// layer all the way down to the SEV platform. Most of these cases have
+/// been enumerated; however, there is a possibility that some error
+/// conditions are not encapsulated here.
+#[derive(Debug)]
+pub enum Indeterminate<T: Debug> {
+    /// The error condition is known.
+    Known(T),
+
+    /// The error condition is unknown.
+    Unknown,
+}
+
+impl<T: Debug + Display> Display for Indeterminate<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Indeterminate::Known(e) => {
+                write!(f, "{}", e)
+            }
+            Indeterminate::Unknown => {
+                write!(f, "Unknown Error",)
+            }
+        }
+    }
+}
+
+impl<T: Debug + Display> error::Error for Indeterminate<T> {}
+
+/// Error conditions returned by the SEV platform or by layers above it
+/// (i.e., the Linux kernel).
+///
+/// These error conditions are documented in the AMD SEV API spec, but
+/// their documentation has been copied here for completeness.
+#[derive(Debug)]
+pub enum Error {
```

Comment:
Should we move this and related into a separate module (e.g. `error.rs`)? Also I wonder if all of these are still relevant, given that some functionality is removed :thinking:. But probably good to have everything for completeness.